### PR TITLE
Mpuncel/protect against farm panic

### PR DESCRIPTION
**Pull request by mpuncel**

### CODE REVIEW

File: pkg/roll/farm.go [hunk 1]
```diff
@@ -134,6 +135,12 @@ START_LOOP:
 					rlLogger.WithError(err).Errorln("Could not read new RC")
 					continue
 				}
+
+				if rcField == nil {
```
Comment (anthonybishopric) [inline pkg/roll/farm.go]: I think I would have preferred modeling this after the `os` package. 

```
if rcs.IsNotExist(err) {
 ...
}
else if err != nil {
 ... 
}
```

Adding a new pointer makes it possible to test for nil in places, but it also makes it hard to use the zero value of the struct safely. Not a serious enough problem to block this PR, just expressing that the `return nil, nil` case from `rcs.Get` deserves something a bit more structurally meaningful.

File: pkg/kp/rcstore/store.go [hunk 2]
```diff
@@ -82,3 +85,11 @@ func RCUpdateLockPath(rcId fields.ID) (string, error) {
 
 	return path.Join(kp.LOCK_TREE, rcPath, "update"), nil
 }
+
+func IsNotExist(err error) bool {
+	if err == NoReplicationController {
```
Comment (petertseng) [inline pkg/kp/rcstore/store.go]: is this function equivalent to `return err == NoReplicationController` or is there something more subtle going on that prevents this?

Comment (mpuncel) [inline pkg/kp/rcstore/store.go]: nah you're right :) it used to be a type assertion on a type and i didn't clean up all the way

File: pkg/kp/rcstore/consul_store.go [hunk 3]
```diff
@@ -341,7 +341,7 @@ func (s *consulStore) Watch(rc *fields.RC, quit <-chan struct{}) (<-chan struct{
 				case <-quit:
 				}
 			} else {
-				*rc = newRC
+				rc = &newRC
```
Comment (petertseng) [inline pkg/kp/rcstore/consul_store.go]: as I understand it, [this will result in no externally-visible change to `rc`](http://play.golang.org/p/BOoyrK0lTa) - is that OK with you? If it is OK, how do callers of this function get an updated RC (or do they not care about having one?)?

**Comment by petertseng** [comment]
I think I'm happy with this. :+1:
